### ISSUE TITLE
AttributeError: module 'lib' has no attribute 'OpenSSL_add_all_algorithms'. AWS sam build failure.

### ISSUE THREAD

**Issue by RichaSharmaTR** [pyca/cryptography#7959]
I'm getting below error after installing latest version of cryptography by using requirements.txt file in AWS deployment.

```
import OpenSSL.SSL
File "/root/.pyenv/versions/3.9.12/lib/python3.9/site-packages/OpenSSL/__init__.py", line 8, in <module>
from OpenSSL import crypto, SSL
File "/root/.pyenv/versions/3.9.12/lib/python3.9/site-packages/OpenSSL/crypto.py", line 3268, in <module>
_lib.OpenSSL_add_all_algorithms()
**_AttributeError: module 'lib' has no attribute 'OpenSSL_add_all_algorithms'_**
```

versions used during installation- 
``cryptography-39.0.0``
 ``Python- 3.9``
``cffi--1.15.1 ``
``pip-22.3.1``
``setuptools-58.1.0``

Could you please help me in resolving this issue?

**Comment by reaperhulk** [comment]
This occurs with an outdated pyOpenSSL version

**Comment by shjasti** [comment]
Hi I'm having the same issue with `cryptography==39.0.0` and `pyOpenSSL==23.0.0`. Downgrading to `cryptography==38.0.4` and `pyOpenSSL==22.1.0` didn't really help either.

**Comment by alex** [comment]
Are you _sure_ you're actually using cryptography 39.0.0 and pyopenssl 23.0.0?

The code to trigger this simply doesn't exist in pyopenssl, so I doubt that's what's happening. Perhaps try re-creating your virtualenv from scratch.

**Comment by shjasti** [comment]
@alex yep my mistake! turns out I was just looking in the wrong place. thanks!

**Comment by zjjott** [comment]
old version pyOpenssl using "cryptography>=35.0" ,but "cryptography==39.0.0" let openssl broken, and then pip broken too.
maybe cryptography should keep compatibility as a cornerstone of pip?

**Comment by reaperhulk** [comment]
For aws-sam-cli users they have fixed their bug in `1.68.0`. I'm going to leave this issue open for a bit longer so others can easily find the solution on the issue tracker (upgrade pyOpenSSL in your pins!), but ultimately this issue is not `cryptography`'s bug.

**Comment by RichaSharmaTR** [comment]
Hi @reaperhulk 
The issue is fixed in the latest version of aws-sam-cli. I am not seeing any build failures now. This bug is good to close now.
Thanks!
